src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java

```diff
@@ -460,6 +463,9 @@ private ProgrammingExerciseStudentParticipation configureRepositoryWebHook(Progr
         if (!participation.getInitializationState().hasCompletedState(InitializationState.INITIALIZED)) {
             versionControlService.get().addWebHook(participation.getRepositoryUrlAsUrl(), ARTEMIS_BASE_URL + PROGRAMMING_SUBMISSION_RESOURCE_API_PATH + participation.getId(),
                     "Artemis WebHook");
+            // Optional webhook from the VCS to the CI (needed for some systems such as GitLab + Jenkins)
+            final var ciHookUrl = continuousIntegrationService.get().getWebhookUrl(participation.getProgrammingExercise().getProjectKey(), participation.getBuildPlanId());
+            ciHookUrl.ifPresent(s -> versionControlService.get().addWebHook(participation.getRepositoryUrlAsUrl(), s, "Artemis trigger to CI", CI_TOKEN));
```

Comment:
why do we need the CI_TOKEN here? I think this is specific to the version control service and should not be exposed to the ParticipationService, please move it into the `addWebHook` method.

I think we could even change the VCS method to `addWebHooks(...)` and then the concrete VCS service decides which webhooks are actually necessary.

What do you think?

Reply:
Goode idea. I moved the creation of webhooks to a new abstract superclass `ArtemisVersionControlService` which calls `addWebHook` for the different repositories. The concrete implementation of `addWebHook` can then be found in the extending class.

Reply:
Hmm, why do we need a new abstract superclass for this?

Reply:
We don't, but this would introduce code duplication since the code for everything except actually sending the request creating the webhook is the same for all concrete implementations.